internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -167,12 +167,14 @@ func (f *FirstReadyWithCapacity) FindCluster(kafka *dbapi.KafkaRequest) (*api.Cl
 
 	// Find first ready cluster that has remaining capacity (total streaming unit used by existing and requested kafka is within the cluster maxUnit limit)
 	for _, cluster := range clusters {
-		currentStreamingUnitsUsed := streamingUnitCountPerRegionList.GetStreamingUnitCountForClusterAndInstanceType(cluster.ClusterID, kafka.InstanceType)
-		capacityInfo := cluster.RetrieveDynamicCapacityInfo()
-		maxStreamingUnits := capacityInfo[kafka.InstanceType].MaxUnits
-
-		if currentStreamingUnitsUsed+instanceSize.CapacityConsumed <= int(maxStreamingUnits) {
-			return cluster, nil
+		if cluster.ClusterType == api.ManagedDataPlaneClusterType.String() {
```

Comment:
Could we refactor this logic to its own method?

Reply:
do you mean something like `cluster.IsEnterpriseType()`?

Reply:
No. This code is inside a loop that processes N clusters each one individually. What I am referring is to extract the logic that deals with the cluster individually to a method that is called from the loop